Sources/TuistGraph/Models/InfoPlist.swift

```diff
@@ -1,7 +1,11 @@
 import Foundation
 import TSCBasic
 
-public enum InfoPlist: Equatable, Codable {
+public protocol PlistTypesProtocol {}
```

Comment:
I wonder if we can take a different approach with this protocol so that [the implementation here](https://github.com/tuist/tuist/pull/5377/files#diff-653ecf513cd273057e59845782f0b9ee211646b5433b186c69219085a54e9393R21) can be done with ` switch` statement to let the compiler warns us if we are missing any type in the codebase that also implements the protocol. What about leveraging the `Plist` enum underneath for that:

```swift
public enum Plist {
  case infoPlist(InfoPlist)
  case entitlements(Entitlements)
}
```
cc @kwridan